src/snapred/backend/data/LocalDataService.py

```diff
@@ -75,9 +76,57 @@ def __init__(self) -> None:
         self.instrumentConfig = self.readInstrumentConfig()
         self.mantidSnapper = MantidSnapper(None, "Utensils")
 
+    ##### MISCELLANEOUS METHODS #####
+
+    def readDetectorState(self, runId: str) -> DetectorState:
+        detectorState = None
+        pvFile = self._readPVFile(runId)
+        try:
+            detectorState = DetectorState(
+                arc=[pvFile.get("entry/DASlogs/det_arc1/value")[0], pvFile.get("entry/DASlogs/det_arc2/value")[0]],
+                wav=pvFile.get("entry/DASlogs/BL3:Chop:Skf1:WavelengthUserReq/value")[0],
+                freq=pvFile.get("entry/DASlogs/BL3:Det:TH:BL:Frequency/value")[0],
+                guideStat=pvFile.get("entry/DASlogs/BL3:Mot:OpticsPos:Pos/value")[0],
+                lin=[pvFile.get("entry/DASlogs/det_lin1/value")[0], pvFile.get("entry/DASlogs/det_lin2/value")[0]],
+            )
+        except:  # noqa: E722
+            raise ValueError(f"Could not find all required logs in file '{self._constructPVFilePath(runId)}'")
+        return detectorState
+
+    ##### FILE METHODS #####
+
     def fileExists(self, path):
         return os.path.isfile(path)
 
+    def _extractFileVersion(self, file: str) -> int:
+        filePattern = ".*/v_(\d+)/.*"
+        result = re.search(filePattern, file)
+        if result is None:
+            raise FileNotFoundError(f"Failed to locate a file version for {file}")
+        else:
+            return int(result.group(1))
```

Comment:
I really did not like relying on the multi-split method, as I found it tended to fail a lot, and when it failed it did so unhelpfully.  I think this is safer.